pkg/ring/kv/etcd/etcd.go

```diff
@@ -34,10 +40,41 @@ func (cfg *Config) RegisterFlagsWithPrefix(f *flag.FlagSet, prefix string) {
 	f.Var((*flagext.StringSlice)(&cfg.Endpoints), prefix+"etcd.endpoints", "The etcd endpoints to connect to.")
 	f.DurationVar(&cfg.DialTimeout, prefix+"etcd.dial-timeout", 10*time.Second, "The dial timeout for the etcd connection.")
 	f.IntVar(&cfg.MaxRetries, prefix+"etcd.max-retries", 10, "The maximum number of retries to do for failed ops.")
+	// Get certs from flag params and set config TLS, if present
+	f.StringVar(&cfg.CertFile, prefix+"etcd.cert-file", "", "The TLS certificate file path")
+	f.StringVar(&cfg.KeyFile, prefix+"etcd.key-file", "", "The TLS private key file path")
+	f.StringVar(&cfg.TrustedCAFile, prefix+"etcd.client-ca-file", "", "The trusted CA file path")
+}
+
+// GetTLS sets the TLS config field with certs
+func (cfg *Config) GetTLS() (*tls.Config, error) {
+	if cfg.CertFile == "" && cfg.KeyFile == "" {
+		return nil, nil
+	}
+	tlsInfo := &transport.TLSInfo{
+		CertFile: cfg.CertFile,
+		KeyFile:  cfg.KeyFile,
+	}
+	if cfg.TrustedCAFile != "" {
+		tlsInfo.TrustedCAFile = cfg.TrustedCAFile
+	}
+	tlsConfig, err := tlsInfo.ClientConfig()
+	if err != nil {
+		return nil, err
+	}
+	// Added below for self-signed certs
+	if cfg.InsecureSkipVerify {
+		tlsConfig.InsecureSkipVerify = true
+	}
+	return tlsConfig, nil
 }
 
 // New makes a new Client.
 func New(cfg Config, codec codec.Codec) (*Client, error) {
+	tlsConfig, err := cfg.GetTLS()
+	if err != nil {
+		return nil, err
```

Comment:
In case an error occurs, we should give an hint to the user to which part of the initialisation fail. The error we may get from `GetTLS()` could be very generic.

```suggestion
		return nil, errors.Wrapf(err, "unable to initialise TLS configuration for etcd")
```